### PR TITLE
Deactivate Account Page and Auto Logout

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Adds UI for Deactivate Account page (Desktop/Mobile Web)
- Adds functionality to log out deactivated users if they're still signed in, and handles sign-ins to deactivated accounts

**Note**: Also fixes `ActionDrawer` for mobile web to have red text for destructive actions in dark mode

In order to get sign in to do the right thing when attempting to sign in deactivated accounts, updates `fetchAccountFailed` to provide an error back to the caller to know why it failed.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
- ❗ Touches existing sign-in/fetch-account flows - should not auto log out users that are not deactivated! ❗ 

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.
- Manually against staging using test accounts toggling back and forth between being deactivated and not (for the UI)
- Tested that tabs that you're already signed in (and therefore have the user cached) still log you out.

TODO:
- [x] Test mobile client is logged out automatically
- [x] Check with Julian that the action drawer is an acceptable alternative to the figma mock
- [x] Build react native drawer
- [x] Add tracking to know how often users deactivate accounts
- [x] Handle error case in deactivation

### How will this change be monitored?

- Added tracking to know how often fetching accounts fails on sign in

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

File: src/containers/deactivate-account-page/DeactivateAccountPage.module.css [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+.list {
+  list-style-type: disc;
+  line-height: 32px;
+  margin-inline-start: 1.5em;
+  margin-bottom: 50px;
+  font-weight: var(--font-demi-bold);
+}
+.list li:last-child {
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/DeactivateAccountPage.module.css:25]: i like this grouping, but i think @sliptype tends to not -- we should agree on a convention / find a good css linter

File: src/containers/deactivate-account-page/DeactivateAccountPage.module.css [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  background-color: var(--accent-red);
+}
+.button.buttonDanger:active {
+  background-color: var(--accent-red-dark-1);
+}
+
+.button > span {
+  font-size: var(--font-m) !important;
+}
+/** Confirm modal **/
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/DeactivateAccountPage.module.css:45]: nit \n

File: src/containers/deactivate-account-page/DeactivateAccountPage.tsx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+            text={messages.buttonGoBack}
+            type={isLoading ? ButtonType.DISABLED : ButtonType.PRIMARY_ALT}
+          />
+        </div>
+      </div>
+    </Modal>
+  )
+}
+
+export const DeactivateAccountPageDesktop = ({
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/DeactivateAccountPage.tsx:155]: let's pull these into 3 components

DeactivateAccountPage.tsx
mobile/DeactivateAccount.tsx
desktop/DeactivateAccount.tsx

and perhaps even follow the DeactivateAccountProvider.tsx model we have set up for something like the profile page.

File: src/containers/deactivate-account-page/store/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+function* handleDeactivateAccount() {
+  try {
+    yield call(waitForBackendSetup)
+    const accountUserId: Nullable<ID> = yield select(getUserId)
+    const userMetadata: ReturnType<typeof getAccountUser> = yield select(
+      getAccountUser
+    )
+    yield put(
+      requestConfirmation(
+        DEACTIVATE_CONFIRMATION_UID,
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/store/sagas.ts:37]: very snazzy -- saw this on the confirmer preview while testing!

File: src/containers/deactivate-account-page/store/sagas.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        function* () {
+          const { blockHash, blockNumber } = yield call(
+            AudiusBackend.updateCreator,
+            { ...userMetadata, is_deactivated: true },
+            accountUserId /* note: as of writing, unused parameter */
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/store/sagas.ts:42]: odd.

File: src/containers/deactivate-account-page/store/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import { getAccountUser, getUserId } from 'common/store/account/selectors'
+import { Nullable } from 'common/utils/typeUtils'
+import AudiusBackend from 'services/AudiusBackend'
+import { waitForBackendSetup } from 'store/backend/sagas'
+import { requestConfirmation } from 'store/confirmer/actions'
+import { confirmTransaction } from 'store/confirmer/sagas'
+import {
+  getConfirmCalls,
+  getConfirmLength,
+  getIsConfirming
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/store/sagas.ts:15]: unused import

File: src/containers/deactivate-account-page/store/sagas.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    )
+  } catch (e) {
+    console.error(e)
+    yield put(deactivateAccountFailed())
+  }
+}
+
+function* waitAndSignOut() {
+  // Wait for all confirmations to end so that reloading doesn't trigger beforeUnload dialog
+  // Note: Waiting for getIsConfirming doesn't appear to work here
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/store/sagas.ts:73]: do you know why it didn't work?
i imagine waitForValue getIsConfirming on false should work?

File: src/containers/deactivate-account-page/store/slice.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+const initialState: DeactivateAccountState = {
+  status: undefined
+}
+
+const slice = createSlice({
+  name: 'application/ui/deactivateAccount',
+  initialState,
+  reducers: {
+    deactivateAccount: state => {
+      state.status = Status.LOADING
```
Comment (raymondjacobson) [inline src/containers/deactivate-account-page/store/slice.ts:18]: 😈  nice immutables!

File: src/store/account/sagas.js [hunk 9]
```diff
@@ -152,23 +152,31 @@ export function* fetchAccountAsync(action) {
...
         cachedAccountUser.orderedPlaylists
       )
       yield put(accountActions.fetchAccountSucceeded(cachedAccount))
-    } else {
-      if (!getCurrentUserExists()) {
-        yield put(accountActions.fetchAccountFailed())
-      }
+    } else if (!getCurrentUserExists()) {
+      yield put(
+        accountActions.fetchAccountFailed({ reason: 'ACCOUNT_NOT_FOUND' })
```
Comment (raymondjacobson) [inline src/store/account/sagas.js:168]: love it

**Review by raymondjacobson — APPROVED** [review]
lovely

File: src/containers/deactivate-account-page/store/sagas.ts (hunk 7 above)
Comment (rickyrombo) [reply src/containers/deactivate-account-page/store/sagas.ts:73]: It stopped checking, unsure why. I'm pretty sure that's exactly what I tried.

File: src/containers/deactivate-account-page/DeactivateAccountPage.tsx (hunk 3 above)
Comment (rickyrombo) [reply src/containers/deactivate-account-page/DeactivateAccountPage.tsx]: done!

File: src/containers/deactivate-account-page/DeactivateAccountPage.module.css (hunk 1 above)
Comment (rickyrombo) [reply src/containers/deactivate-account-page/DeactivateAccountPage.module.css:25]: @sliptype   🤜  FIGHT 🤛  FIGHT 👊  FIGHT

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/mjp-deactivate-account-ui

File: src/containers/deactivate-account-page/store/sagas.ts (hunk 5 above)
Comment (rickyrombo) [reply src/containers/deactivate-account-page/store/sagas.ts:42]: Yeah we seem to pull `user_id` from the `userMetadata` instead

File: src/containers/deactivate-account-page/store/sagas.ts (hunk 7 above)
Comment (rickyrombo) [reply src/containers/deactivate-account-page/store/sagas.ts:73]: Yeah just tried again and console logged in the validator, and it only checks once and at that point returns true, then the "take" never goes again I guess and you're stuck waiting forever.

File: src/containers/deactivate-account-page/DeactivateAccountPage.module.css (hunk 1 above)
Comment (sliptype) [reply src/containers/deactivate-account-page/DeactivateAccountPage.module.css:25]: I have no dog in this fight

File: src/components/tracks-table/TableOptionsButton.js [hunk 10]
```diff
@@ -48,7 +48,7 @@ class TableOptionsButton extends Component {
         extraMenuItems: onRemove ? [removeMenuItem] : []
       }
     }
-    if (isOwnerDeactivated && !onRemove && !isFavorited && !isReposted) {
+    if (isOwnerDeactivated && !onRemove && !isFavorited) {
```
Comment (raymondjacobson) [inline src/components/tracks-table/TableOptionsButton.js:51]: nice

**Review by raymondjacobson — APPROVED** [review]

**Pull request merged**